### PR TITLE
Herc20 hbit refund

### PR DESCRIPTION
**Pull request by luckysori**
Almost exclusively TS changes after a little bit of debugging.

I would be happy to refine some e2e test suite APIs, because I probably modified things too fast.

Also, I believe there is no way to "refund" Halight, so I don't know if I should assert anything about lightning (or bitcoin) at the end of `alice-refunds-bob-refunds` tests.

### CODE REVIEW

File: api_tests/src/actors/actor.ts [hunk 1]
```diff
@@ -425,6 +425,19 @@ export class Actor {
         await this.assertRedeemed();
     }
 
+    /**
+     * Wait for and execute the refund action
+     */
+    public async refund() {
```
Comment (D4nte) [inline api_tests/src/actors/actor.ts:431]: 📣 use `assertAndExecuteNextAction` instead, do not add this new method.

File: api_tests/src/actors/actor.ts [hunk 2]
```diff
@@ -630,6 +669,33 @@ export class Actor {
...
+                assetKey,
+                maximumFee
+            );
+            const expectedBalance = this.startingBalances.get(assetKey);
+            const currentWalletBalance = await wallet.getBalanceByAsset(
+                defaultAssetValue(asset, ledger)
+            );
+            const balanceInclFees = expectedBalance - maximumFee;
+            expect(currentWalletBalance).toBeGreaterThanOrEqual(
+                // @ts-ignore: Jest supports bigint, types to be fixed updated with
```
Comment (D4nte) [inline api_tests/src/actors/actor.ts:692]:
```suggestion
```

This is not needed any more.

File: api_tests/src/actors/actor.ts (hunk 1 above)
Comment (D4nte) [reply api_tests/src/actors/actor.ts:431]: Also, in `hbit_herc20.ts`, do not use `redeem`, etc instead use `assertAndExecuteNextAction`.

Comment (luckysori) [reply api_tests/src/actors/actor.ts:431]: Oh, yeah. Nice! I'll remove all these actions. We weren't actually using them after the last patch.

File: api_tests/src/actors/swap_factory.ts [hunk 3]
```diff
@@ -23,16 +23,16 @@ import { HbitRequestParams } from "comit-sdk/dist/src/cnd/swaps_payload";
 
 declare var global: HarnessGlobal;
 
-interface Ledgers {
-    alpha: keyof AllWallets;
-    beta: keyof AllWallets;
+interface SwapSettings {
+    ledgers?: { alpha: keyof AllWallets; beta: keyof AllWallets };
+    instantRefund?: boolean;
```
Comment (thomaseizinger) [inline api_tests/src/actors/swap_factory.ts:28]: :joy:

Comment (luckysori) [reply api_tests/src/actors/swap_factory.ts:28]: You hurt me, @thomaseizinger :sob:

Comment (thomaseizinger) [reply api_tests/src/actors/swap_factory.ts:28]: Sorry, I didn't mean to :)
I was just positively amused!

**Pull request merged**